eZ/Publish/Core/Search/Common/Exception/InvalidFullTextSearchString.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace eZ\Publish\Core\Search\Common\Exception;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+
+/**
+ * Exception thrown FullText search string is invalid.
+ */
+class InvalidFullTextSearchString extends InvalidArgumentException
```

Comment:
I'd say either keep it as is or `InvalidFullTextSearchQuery`.

Side thing: if we intend for external bundle to be able to catch it (and thus refer to it), shouldn't it be a part of API, not Core? Or maybe SPI to avoid cluttering API namespace? POV @andrerom?